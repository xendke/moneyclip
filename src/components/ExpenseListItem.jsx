import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = (props) => (
	props.id ?
		(<div>
			<Link to={`/edit/${props.id}`}>
				<h4>{props.description}</h4>
			</Link>
			<p>
				{numeral(props.amount / 100).format('$0,0.00')}
				-
				{moment(props.createdAt).format("MMMM Do YYYY")}
			</p>
		</div>
		) : (
			<p>Invalid Expense</p>
		)

);

export default ExpenseListItem;