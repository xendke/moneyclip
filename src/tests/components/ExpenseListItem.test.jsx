import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

it('renders ExpenseListItem given an expense', () => {
	const wrapper = shallow(<ExpenseListItem expense={expenses[0]} />);

	expect(wrapper).toMatchSnapshot();
});

it('renders ExpenseListItem given no expense', () => {
	const wrapper = shallow(<ExpenseListItem />);

	expect(wrapper).toMatchSnapshot();
});