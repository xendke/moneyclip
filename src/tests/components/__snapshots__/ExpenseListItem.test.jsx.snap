// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ExpenseListItem given an expense 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h4>
      the quick brown fox jumped over
    </h4>
  </Link>
  <p>
    $1.00
    -
    December 31st 1969
  </p>
</div>
`;

exports[`renders ExpenseListItem given no expense 1`] = `
<p>
  Invalid Expense
</p>
`;
